content/blog/2016/2016-05-31-Jenkins-Usage-Statistics-Analysis-Intro.adoc

```diff
@@ -0,0 +1,98 @@
+Understanding the trend in upgrades and downgrades have lots of utilities, some of them have already been explained
+earlier which includes measuring the popularity, spotting downgrades, giving warning about the wrong versions quickly etc.
+
+===== Usecase 1.1: Upgrading time metrics
+This analysis refers to the time taken by the users to upgrade its jenkins installation and its plugins. Required dataset
+for the analysis are link:https://bigquery.cloud.google.com/table/jenkins-user-stats:jenkinsstats.jenkins_usage?_ga=1.195703460.947004610.1452882463&pli=1[jenkins-usage] data and link:http://mirror.xmission.com/jenkins/updates/current/release-history.json[release-history] data. Types of queries that are supported in this
+analysis are:
+* *Query1 :* Enter the update release date, whose performance user wants to study.
+* *Query2 :* Enter the plugin name, whose release history, charts and upgrading time metric, user wants to study.	
+
+===== Usecase 1.2: Spotting dowgrades
+Here we are interested to know, how many installations downgraded from any given version to any earlier version. The idea
+is to measure the monotonicity of the version_number vs timestamp graph per installation id because graph must be
+increasing if there are no downgrades.
+
+===== Usecase 1.3: Correlation with the perceived quality of Jenkins release 
+To correlate what users are saying to what users are doing, we have community ratings which tells us about the ratings and
+reviews of the releases and has following parameters:			
+* Used the release on production site w/o major issues.		
+* Don’t recommend to other.			
+* Tried but rolled it back to the previous version. 
+Similar parameters can be fetch from the analysis explained in above section.
+
+==== Usecase 2: Plugin Recommendation System
+This section involves setting up ground work for the plugin recommendation system. The idea is to find out the set of
+plugins which are most likely to be used together. Here we will be following both content based filtering as well as
+collaborative filtering approach. 
+
+===== Collaborative Filtering
+This approach is based upon the collecting and analysing large amount of information on user’s behaviours, activities or
+preferences. We have implicit form of the data about the plugins, that is for every install ids, we know the set of
+plugins installed. We can exploit this information to construct plugin usage graph where nodes are the plugins and the
+edges between them is the number of installations in which both plugins are installed together.
+
+===== Content-based Filtering
+Content-based filtering methods are based on a description of the item and a profile of the user’s preference for example
+recommending items that are similar to the those that a user liked in the past or examining in the present. Here, we are
+exploiting link:http://stats.jenkins-ci.org/jenkins-stats/jenkinsgraph.html?filter=kohsuke[Jenkins plugin dependency graph]
```

Comment:
This blog post is supposed to stand alone and be understandable by people who know nothing about your project.